### DISCUSSION TITLE
icons a la Cmake as commands shortcuts?

### DISCUSSION THREAD

**Discussion by ThomasCartier** [rust-lang/rust-analyzer#15884] [Q&A]
Hi,

Coding in C++ with cmake inside VSCode, you have the luxury of a project selector, build icon, debug icon and run icon in the status bar.

You are one click away from your command.

Do you plan to at least add an icon for "cargo check", "fix" and "build" in the status bar ?

That would be very convenient and faster, instead of F1 or terminal typing.

A slight productivity bump.

Thanks,

**Comment by Veykril** [comment]
Do you have a sample screenshot of this at hand? Alternatively, which extension(s) offer this for C++ so that we could take a look at those?

**Reply by lnicola** [reply]
Something like:

![image](https://github.com/rust-lang/rust-analyzer/assets/308347/436f97d3-9f82-4cc8-9112-c3eafbff6ae9)

Most of them will open a drop-down in the command palette. Run does nothing for me at the moment.

We could even add Clippy in there for running it just once (instead of toggling `checkCommand`), switch to offline mode, disable proc macros and build scripts and so on.

**Reply by ThomasCartier** [reply]
Yes at first, when setting things up, but then it's direct.
This is my experience after years of C++ in VSCode
